back/engines/commercial/granular_permissions/spec/acceptance/permissions_spec.rb

```diff
@@ -30,10 +30,53 @@
       end
 
       example_request 'List all permissions of a project' do
-        expect(status).to eq(200)
-        json_response = json_parse(response_body)
+        assert_status 200
+        json_response = json_parse response_body
         expect(json_response[:data].size).to eq Permission.available_actions(@project).size
       end
+
+      example_request 'List all permissions efficiently include custom fields', document: false do
+        permission = @project.permissions.first
+        field2 = create :custom_field
+        field1 = create :custom_field
+        field3 = create :custom_field
+        field1.move_to_top
+        field2.reload
+        field3.reload
+        permission.permissions_custom_fields.create!(custom_field: field2, required: true)
+        permission.permissions_custom_fields.create!(custom_field: field1, required: true)
+        permission.permissions_custom_fields.create!(custom_field: field3, required: true)
```

Comment:
Would it be useful to include a not required field too?

Reply:
Sure, I'll do that :)